tests/containers/container_engine.pm

```diff
@@ -32,6 +32,89 @@ use containers::common;
+    assert_script_run("$runtime container restart basic_test_container");
+    validate_script_output("$runtime ps", qr/basic_test_container/);
+    validate_script_output("$runtime container inspect --format='{{.State.Running}}' basic_test_container", qr/true/);
+
+    ## Test logs
+    assert_script_run("$runtime run -d --name logs_test $image echo 'log test canary string'");
+    # retry because it could be that the log is not yet collected after the previous command completes
+    validate_script_output_retry("$runtime logs logs_test", qr/log test canary string/, retry => 3, delay => 60);
+    assert_script_run("$runtime container stop logs_test");
+    assert_script_run("$runtime container rm logs_test");
+
+    ## Test exec and image creation
+    assert_script_run("$runtime container exec basic_test_container touch /canary");
+    assert_script_run("$runtime container commit basic_test_container example.com/tw-commit_test");
+    validate_script_output("$runtime image ls --all", qr?example.com/tw-commit_test?);
+    validate_script_output("$runtime image ls --all", qr?example.com/prune-test?);
+    assert_script_run("$runtime run --rm example.com/tw-commit_test stat /canary", fail_message => "canary file not present in generated container");
+    assert_script_run("$runtime image rm example.com/tw-commit_test");
+
+    ## Test connectivity inside the container
+    assert_script_run("$runtime container exec basic_test_container curl -sfI https://opensuse.org", fail_message => "cannot reach opensuse.org");
+
+    ## Test `--init`, i.e. the container process won't be PID 1 (to avoid zombie processes)
+    # expected output: the `ps` command is not running as PID 1.
+    validate_script_output("$runtime run --rm --init $image ps --no-headers -xo 'pid args'", sub { $_ !~ m/ 1 .*ps.*/ });
+
+    ## Test prune
+    assert_script_run("$runtime container commit basic_test_container example.com/prune-test");
+    validate_script_output("$runtime image ls --all", qr?example.com/prune-test?);
+    assert_script_run("$runtime image prune -af");
+    validate_script_output("$runtime ps", sub { $_ !~ m?example.com/prune-test? });
+    validate_script_output("$runtime image ls", qr/tumbleweed/, fail_message => "Tumbleweed image removed, despite being in use");
+    assert_script_run("$runtime system prune");
+    validate_script_output("$runtime image ls", qr/tumbleweed/, fail_message => "Tumbleweed image removed, despite being in use");
+    assert_script_run("! $runtime rmi -a");    # should not be possible because image is in use
+
+    ## Removing containers
+    assert_script_run("$runtime container stop basic_test_container");
+    assert_script_run("$runtime container rm basic_test_container");
+    validate_script_output("$runtime container ls --all", sub { $_ !~ m/basic_test_container/ });
+
+    ## Note: Leave the tumbleweed container to save some bandwidth. It is used in other test modules as well.
```

Comment:
We could add a docker save, docker rmi, docker load here.

Reply:
I've added already `image rm` checks and would not add `save` and `load` yet, because it was not part of the original PR.